packages/core/src/area-chart.ts

```diff
@@ -0,0 +1,176 @@
+// D3 Imports
+import { select, selectAll, mouse } from "d3-selection";
+import { area } from "d3-shape";
+
+import { ScatterChart } from "./scatter-chart";
+import * as Configuration from "./configuration";
+import { ChartConfig, AreaChartOptions, ChartType } from "./configuration";
+
+import { getD3Curve } from "./services/curves";
+import { Tools } from "./tools";
+
+export class AreaChart extends ScatterChart {
+	areaGeneratorFill: any;
+	areaGeneratorStroke: any;
+
+	options: AreaChartOptions;
+
+	constructor(holder: Element, configs: ChartConfig<AreaChartOptions>) {
+		super(holder, configs);
+
+		// initialize options
+		if (configs.options) {
+			this.options = Tools.merge({}, Configuration.options.AREA, configs.options);
+		} else {
+			this.options = Tools.merge({}, Configuration.options.AREA);
+		}
+
+		this.options.type = ChartType.AREA;
+	}
+
+	draw() {
+		this.innerWrap.style("width", "100%")
+			.style("height", "100%");
+
+		const { line: margins } = Configuration.charts.margin;
+
+		this.innerWrap.style("width", "100%").style("height", "100%");
+
+		this.innerWrap.attr("transform", `translate(${margins.left}, ${margins.top})`);
+
+		let curveName;
+		let curveOptions;
+		this.options.curve = this.options.curve || "curveLinear";
+		if (typeof this.options.curve === "string") { // curve: 'string'
+			curveName = this.options.curve;
+			curveOptions = {};
+		} else { // curve: { name: 'string' }
+			curveName = this.options.curve.name || "curveLinear";
+			curveOptions = this.options.curve;
+			delete curveOptions["name"];
+		}
+
+		// D3 area generator function
+		this.areaGeneratorFill = area()
+			.x((d, i) => this.x(this.displayData.labels[i]) + this.x.step() / 2)
+			.y1(this.y(0))
+			.y0((d: any) => this.y(d))
+			.curve(getD3Curve(curveName, curveOptions));
+
+		this.areaGeneratorStroke = area()
+			.x((d, i) => this.x(this.displayData.labels[i]) + this.x.step() / 2)
+			.y((d: any) => this.y(d))
+			.curve(getD3Curve(curveName, curveOptions));
+
+		const gLines = this.innerWrap.selectAll("g.lines")
+			.data(this.displayData.datasets)
+			.enter()
+				.append("g")
+				.classed("lines", true);
+
+		gLines.append("path")
+			.attr("stroke", d => this.getStrokeColor(d.label))
+			.datum(d => d.data)
+			.attr("class", "line")
+			.attr("data-area-component", "stroke")
+			.attr("d", this.areaGeneratorStroke);
+
+		gLines.append("path")
+			.attr("fill", d => this.getStrokeColor(d.label))
+			.attr("fill-opacity", 0.3)
+			.datum(d => d.data)
+			.attr("class", "line")
+			.attr("data-area-component", "fill")
```

Comment:
Since the distinction is made/executed in js, I figured an attribute would be more appropriate. That is, I think it'd be weird to use a CSS class but never use/define it in our CSS files.

Reply:
yes we don't apply any styles to the area fill paths currently, but what if a user wanted to?

Reply:
If they wanted to, it can still be selected in the css, which would override the semantic svg attribute: https://github.com/carbon-design-system/carbon-charts/blob/52d1d43771c8c8579c7b0e4d565cb1317741a695/packages/core/src/styles/style.scss#L75

Reply:
If they wanted to, it can still be selected in the css, which would override the semantic svg attribute: https://github.com/carbon-design-system/carbon-charts/blob/52d1d43771c8c8579c7b0e4d565cb1317741a695/packages/core/src/styles/style.scss#L75

Reply:
I would also argue that the fills are content (they correspond to the key) and are thus attributes, not just presentation that would use CSS.

Reply:
I'm only talking about `.attr("data-area-component", "fill")` there, not the other attributes applied

Reply:
resolved in https://github.com/carbon-design-system/carbon-charts/pull/315/commits/046f0a0279ca61b4ca908d895a3bebeaab8c6953